api/src/models/ImageFile.ts

```diff
@@ -56,29 +62,98 @@ class ImageFile {
         var path = require('path');
         var dir = path.dirname(this.filename);
         var filename = this.basename(this.filename);
-        return dir + "/" + filename + "." + size + "." + extension.toLowerCase();
+        return dir + "/" + filename + "/" + size + "/" + filename + "." + extension.toLowerCase();
     }
 
-    ocr() {
-        var txt = this.derivativePath('OCR-DIRTY', 'txt');
+    async ocr() {
+        //TODO: update the following (derivativepath requires size now)
+        var txt = this.derivativePath('OCR-DIRTY', 'txt'); 
         let fs = require('fs');
-        var tesseract = require("node-tesseract-ocr");
-        if (fs.existsSync(txt)){
+        let { exec } = require("child_process");
+        let deriv = await this.ocrDerivative();
+        if (!fs.existsSync(txt)){
             var path = this.basename(txt);
+            let ts_cmd = this.config().tesseractPath() + " " + deriv + " " + txt.slice(0, -4) + " " + this.ocrProperties();
+            exec(ts_cmd, (error, stdout, stderr) => {
+                if (error) {
+                    throw `error: ${error.message}`;
+                    return;
+                } 
+                if (stderr) {
+                    throw `stderr: ${stderr}`;
+                    return;
+                }
+                if (!fs.existsSync(txt)) {
+                    throw "Problem running textcleaner";
+                }
+                console.log(`stdout: ${stdout}`)
+            });
+
         }
 
     }
 
-    ocrDerivative() {
-
+    async ocrDerivative() {
+        let fs = require("fs");
+        let png = this.derivativePath('ocr/pngs', 'png');
+        let { exec } = require("child_process");
+        let deriv = await this.derivative('LARGE');
+        if (!fs.existsSync(png)) {
+            let tc_cmd = this.config().textcleanerPath() + " " + this.config().textcleanerSwitches() + " " + deriv + " " + png;
+            exec(tc_cmd, (error, stdout, stderr) => {
+                if (error) {
+                    throw `error: ${error.message}`;
+                    return;
+                } 
+                if (stderr) {
+                    throw `stderr: ${stderr}`;
+                    return;
+                }
+                if (!fs.existsSync(png)) {
+                    throw "Problem running textcleaner";
+                }
+                console.log(`stdout: ${stdout}`)
+            });
+        }
+        return png;
     }
 
     ocrProperties() {
-
+        let fs = require("fs");
+        var path = require('path');
+        let file = path.dirname(this.filename) + '/ocr/tesseract.config';
+        let dir = path.dirname(file);
+        let content = 'tessedit_char_whitelist ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789.,;:!\'"()&$%-+=[]?<>' + "\xE2\x80\x9C\xE2\x80\x9D\xE2\x80\x98\xE2\x80\x99";
+        if (!fs.existsSync(file)) {
+            fs.writeFile(dir + '/ocr/tesseract.config', content, err => {
```

Comment:
I think you can just use `file` here instead of `dir + '/ocr/tesseract.config'`  -- right now you seem to be taking the string apart and then putting it back together again, but it should be possible to do less work. :-)